vlab-research/fly
# Feature/websurvey routing logic

### What changes have you made?
- refactored such that you can now access a question by manually navigating to /{REF}.
- created a utils folder with a directory of helper functions including `isLast`, `getIndex` and `getNextRef`
- implemented Nandan's solution [on this branch](https://github.com/vlab-research/fly/commit/1199ab5813d43e5b19ebeb36c505a4693efcd5e1) 
- using Svelte's reactivity, the field is now bound to the `ref` 
- fixed a bug where consecutive fields of the same `type` were not reacting to changes in `ref`
- included thankyou screen logic
- replaced the `setTimeout` solution with Svelte's `onMount` hook

### Which issue(s) does this PR fix?
#49 

### How to test
- go to the basepath and check that the first `ref` and the first `field` displays
- use the form button to navigate linearly through the form making sure that the `ref` corresponds to the `field`
- on form submit of the last question a thankyou screen should display
